provider/core/src/any.rs

```diff
@@ -398,10 +388,9 @@ where
     }
 }
 
-impl<M, P> DynamicDataProvider<M> for DowncastingAnyProvider<'_, P>
+impl<M: KeyedDataMarker, P: AnyProvider + ?Sized> DynamicDataProvider<M>
```

Comment:
Issue: This was previously bounded on `DataMarker` but you changed it to `KeyedDataMarker`